rpc/legacyrpc/methods.go

```diff
@@ -3074,6 +3077,76 @@ func signRawTransactions(s *Server, icmd interface{}) (interface{}, error) {
 	return &dcrjson.SignRawTransactionsResult{Results: toReturn}, nil
 }
 
+// makeChangeSourceFromAddress creates a ChangeSource which is used to
+// receive all correlated previous input value.
+func makeChangeSourceFromAddress(address string) txauthor.ChangeSource {
+	return func() ([]byte, uint16, error) {
+		destinationAddress, err := dcrutil.DecodeAddress(address)
+		if err != nil {
+			return nil, 0, err
+		}
+		script, err := txscript.PayToAddrScript(destinationAddress)
+		if err != nil {
+			return nil, 0, err
+		}
+
+		return script, txscript.DefaultScriptVersion, err
+	}
+}
+
+// sweepAccount handles the sweepaccount command.
+func sweepAccount(s *Server, icmd interface{}) (interface{}, error) {
+	cmd := icmd.(*dcrjson.SweepAccountCmd)
+	w, ok := s.walletLoader.LoadedWallet()
+	if !ok {
+		return nil, &ErrUnloadedWallet
+	}
+
+	// use provided fee per Kb if specified
+	feePerKb := txrules.DefaultRelayFeePerKb
+	if cmd.FeePerKb != nil {
+		feePerKb = dcrutil.Amount(*cmd.FeePerKb)
```

Comment:
use dcrutil.NewAmount to convert from a floating point amount between 0-21M to the dcrutil.Amount (in atoms).